include/ginkgo/core/config/property_tree.hpp

```diff
@@ -0,0 +1,209 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_CONFIG_PROPERTY_TREE_HPP_
+#define GKO_PUBLIC_CORE_CONFIG_PROPERTY_TREE_HPP_
+
+
+#include <cassert>
+#include <cstdint>
+#include <exception>
+#include <limits>
+#include <map>
+#include <stdexcept>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+
+namespace gko {
+namespace config {
+
+
+/**
+ * pnode is to describe the property tree.
```

Comment:
```suggestion
 * pnode describes a tree of properties.
 *
 * A pnode can either be empty, hold a value (an string, integer, real, or bool),
 * contain an array of pnode., or contain an mapping between strings and pnodes.
```
I think a little more detail would help.